src/main/java/org/altbeacon/beacon/service/scanner/CycledLeScanner.java

```diff
@@ -40,7 +42,10 @@
     private long mScanPeriod;
 
     protected long mBetweenScanPeriod;
-    protected final Handler mHandler = new Handler();
+
+    protected final Handler mHandler = new Handler(Looper.getMainLooper());
+    protected final Handler mScanHandler;
```

Comment:
@davidgyoung see my answer below (and this [doc](https://developer.android.com/reference/android/os/Handler.html#Handler())), this is the change that makes the whole PR work: by creating it with `new Handler()` you'd be using the `Looper` from the thread that's instantiating `CycledLeScanner`, which is the main thread.

Instead, by creating a `HandlerThread` (which is just a thread with a Looper attached to it) we make sure the "expensive" Bluetooth code from the SDK is posted and executed in our own worker thread.